llama-index-core/llama_index/core/workflow/decorators.py

```diff
@@ -44,15 +45,18 @@ def decorator(func: Callable) -> Callable:
             )
 
         # This will raise providing a message with the specific validation failure
-        event_name, event_types, return_types = validate_step_signature(func)
+        spec = inspect_signature(func)
+        validate_step_signature(spec)
```

Comment:
decoupling introspection from validation seems more future proof, it wasn't actually needed but in the end it simplified the validation logic